### PR TITLE
[SYCL][DebugInfo] Reinstate source locations for some kernel instruct…

### PR DESCRIPTION
**Pull request by stevemerr**
…ions

Now that the source location of the kernel caller function is being assigned to the initialization code in the kernel body, stop clearing the source locations for some other instructions (e.g. call to operator() and the return instruction).

Signed-off-by: Steve Merritt <steve.merritt@intel.com>

### CODE REVIEW

**Review by premanandrao — DISMISSED** [review]
Looks okay to me.

**Review by bwyma — COMMENTED** [review]
LGTM.

File: clang/test/CodeGenSYCL/debug-info-srcpos-kernel.cpp [hunk 1]
```diff
@@ -28,17 +28,16 @@ int main() {
...
-// CHECK: [[LINE_B0]] = !DILocation(line: 0
+// CHECK: [[LINE_B0]] = !DILocation(line: 16,{{.*}}scope: [[BLOCK:![0-9]+]]
+// CHECK: [[BLOCK]] = distinct !DILexicalBlock(scope: [[KERNEL]]
+// CHECK: [[LINE_C0]] = !DILocation(line: 17,{{.*}}scope: [[KERNEL]]
 
-// TODO: [[LINE_B0]] should be mapped to line 15 as well. That said,
-// this 'line 0' assignment is less problematic as the lambda function
-// call would be inlined in most cases.
 // TODO: SYCL specific fail - analyze and enable
 // XFAIL: windows-msvc
```
Comment (bader) [inline clang/test/CodeGenSYCL/debug-info-srcpos-kernel.cpp:43]:
```suggestion
```

**Comment by bader** [comment]
@stevemerr, please, file a tracker for "LLVM_SPIRV :: capability-arbitrary-precision-integers.ll" failure. I don't think it's related to this PR, so I'll merge it.
Tagging @AlexeySotkin.

**Comment by stevemerr** [comment]
@bader @AlexeySotkin - I think the failure was already reported in CMPLRLLVM-22775.  That Jira is closed however there is still an open PR: https://github.com/intel/llvm/pull/2454

**Pull request merged**